storagedriver/s3/s3.go

```diff
@@ -642,6 +642,15 @@ func (d *Driver) URLFor(path string, options map[string]interface{}) (string, er
 		return "", storagedriver.InvalidPathError{Path: path}
 	}
 
+	methodString := "GET"
+	method, ok := options["method"]
+	if ok {
+		if method != "GET" && method != "HEAD" {
+			return "", storagedriver.ErrUnsupportedMethod
+		}
+		methodString, _ = method.(string)
```

Comment:
This will panic if the method is not a string. It should be something like this:

```
m, ok  = method.(string)
if !ok || (m != "GET" && m != "HEAD") {
	return "", storagedriver.ErrUnsupportedMethod
}
methodString = m
```